test/conformance/api/v1/service_test.go

```diff
@@ -478,8 +491,9 @@ func TestServiceWithTrafficSplit(t *testing.T) {
 			Tag:     "candidate",
 			Percent: ptr.Int64(50),
 		}, {
-			Tag:     "latest",
-			Percent: nil,
+			Tag:            "latest",
+			LatestRevision: ptr.Bool(true),
```

Comment:
I don't quite follow why we need this, and needing this seems to conflict with the spec saying "latestRevision may be optionally provided" which I think means a user isn't required to provide this field. My reading of this paragraph in the spec is that latestRevision is always true (whether provided or not) when RevisionName is empty (and vice versa), but you can specify it if you like. I guess it's possible to read it another way, but it seems like something that'd be clearer to correct in the spec, rather than effectively making this field required when it can be trivially inferred? 

> latestRevision may be optionally provided to indicate that the latest ready Revision of the Configuration should be used for this traffic target. **When provided latestRevision MUST be true if revisionName is empty, and it MUST be false when revisionName is non-empty**.

Out of interest if a revisionName is not provided and the optional LatestRevision is not set, what does Cloud Run do?

Reply:
Cloud Run rejects the request to ask either input a revisionName or set LatestRevision to true.

My reading of that paragraph is that **if provided**, latestRevision MUST be true if revisionName is empty, and it MUST be false when revisionName is non-empty. It doesn't means the default value of `latestRevision`.

@evankanderson WDYT?

Reply:
I agree that "if provided" those things are true, I don't agree - fwiw, to my reading anyway - the spec says an implementation can _require_ a user to provide it (otherwise what would "may optionally provide" mean?)

Reply:
I believe that the spec is trying to say "either `latestRevision` is set, or `revisionName` must be set".

What's the expected behavior if neither field is set? That feels like it should be an error.

Reply:
To clarify, do you mean set explicitly?

If neither field is set explicitly, currently Knative serving defaults `latestRevision` to true while Cloud Run throws an error.

Reply:
I'd suggest we clarify this because my reading of this is very much that the spec is saying you can pass latestRevision if you like, but you don't need to. It also seems like using the existing knative open source behaviour (if revisionName is empty latestRevision is implied) maximises compatibility, which seems like quite an important goal for a spec - otherwise we'll have plenty of KSVCs that work in knative open source and not in cloud run.

Reply:
I think setting defaults according to the input `revisionName` makes sense to me. I have dropped the change.